tracing-appender/src/rolling.rs

```diff
@@ -4,12 +4,55 @@ use std::fmt::Debug;
 use std::io;
 use std::path::Path;
 
+/// A file appender with the ability to rotate log files at a fixed schedule.
+///
+/// `RollingFileAppender` implements [`std:io::Write` trait][write] and will block on write operations.
+/// It is intended to be used with [`NonBlocking`][non-blocking] to perform writes without
+/// blocking the current thread.
+///
+/// [write]: https://doc.rust-lang.org/nightly/std/io/trait.Write.html
+/// [non-blocking]: ../non_blocking/struct.NonBlocking.html
+///
+/// # Examples
+///
+/// ```rust
+/// # fn docs() {
+/// let file_appender = tracing_appender::rolling::hourly("/some/directory", "prefix.log");
+/// let (non_blocking, _guard) = tracing_appender::non_blocking(file_appender);
+/// # }
```

Comment:
Maybe we should _just_ show the file appender here (and not the non-blocking writer). We could have an example in `lib.rs` of using both together.

Also, I'd like to show actually building a subscriber that uses the appender, rather than just constructing it.

Reply:
I've removed the non_blocking, but we can't use the file appender with a subscriber on it's own since we don't have a MakeWriter impl for it as you mentioned in a later comment.